### PR TITLE
Add new new method to IContentManager

### LINKED ISSUE #13196: Add GetAsync(IEnumerable<string> contentItemIds, VersionOptions options) to IContentManager
### Is your feature request related to a problem? Please describe.
Currently we have `GetAsync(IEnumerable<string> contentItemIds, bool latest = false)` and `GetAsync(string contentItemId, VersionOptions options)` but not `GetAsync(IEnumerable<string> contentItemIds, VersionOptions options)`. 

When doing bulk update, it is much more helpful to use `GetAsync(IEnumerable<string> contentItemIds, VersionOptions options)` than `GetAsync(IEnumerable<string> contentItemIds, bool latest = false)`. Like if we want to get multiple `VersionOptions.DraftRequired` not just the latest.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #13196

### CODE REVIEW

**Comment by Skrypt** [comment]
So many changes in there. Everything will need to be carefully scrutinized to make sure it doesn't break anything for anyone. The default content manager is a really sensitive piece of code that we generally try to not modify without really thinking about it twice and twice a second time. 😄

**Comment by MikeAlhayek** [comment]
@Skrypt Yes I agree. At least one more person will need to download the PR and try it out. The main change is adding the new method. Then used removed the body of the existing GetAsync(string id, VersionOptions) and instead called the new method just so we have the code in 1 place instead of 2 places.  The code comparison is really hard to follow for some reason in this PR. Maybe if you have some time, you can test drive it?

**Comment by Skrypt** [comment]
Can't commit to doing it but will if I can.

File: src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs [hunk 1]
```diff
@@ -422,6 +526,66 @@ protected async Task<ContentItem> BuildNewVersionAsync(ContentItem existingConte
             return context.BuildingContentItem;
         }
 
+        protected async Task<IEnumerable<ContentItem>> BuildNewVersionsAsync(IEnumerable<ContentItem> existingContentItems)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs:529]: `protected async Task<IEnumerable<ContentItem>> BuildNewVersionAsync(params ContentItem[] existingContentItems)`

File: src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs [hunk 2]
```diff
@@ -246,6 +244,112 @@ public async Task<ContentItem> GetAsync(string contentItemId, VersionOptions opt
             return contentItem;
         }
 
+        public async Task<IEnumerable<ContentItem>> GetAsync(IEnumerable<string> contentItemIds, VersionOptions options)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs:247]: Add a local ToArray()

**Review by sebastienros — APPROVED** [review]
